portal/portal-service-impl/impl/src/java/org/sakaiproject/portal/service/BullhornServiceImpl.java

```diff
@@ -260,11 +263,16 @@ public void update(Observable o, final Object arg) {
                                 Site site = siteService.getSite(siteId);
                                 String title = assignment.getTitle();
                                 String url = assignmentService.getDeepLink(siteId, assignmentId);
+                                Set<String> groupIds = assignment.getGroups();
+                                Collection<String> groupsUsers = authzGroupService.getAuthzUsersInGroups(groupIds);
                                 // Get all the members of the site with read ability
-                                for (String  to : site.getUsersIsAllowed(AssignmentServiceConstants.SECURE_ACCESS_ASSIGNMENT)) {
-                                    if (!from.equals(to) && !securityService.isSuperUser(to)) {
-                                        doAcademicInsert(from, to, event, ref, title, siteId, e.getEventTime(), url);
-                                        countCache.remove(to);
+                                for (String to : site.getUsersIsAllowed(AssignmentServiceConstants.SECURE_ACCESS_ASSIGNMENT)) {
+                                    //  If this is a grouped assignment, is 'to' in one of the groups?
+                                    if (groupIds.size() == 0 || (groupIds.size() > 0 && groupsUsers.contains(to))) {
```

Comment:
An assignment assigned to groups would look like:
Assignment.Access.GROUP.equals(assignment.getTypeOfAccess())

Reply:
But getGroups is still a good indicator of visibility, right? I've still got to get the groups to see who is a member.